include/experimental/__p0009_bits/extents.hpp

```diff
@@ -439,7 +439,12 @@ template <class IndexType, size_t... Extents> class extents {
            sizeof...(OtherIndexTypes) == m_rank_dynamic)))
   MDSPAN_INLINE_FUNCTION
   constexpr explicit extents(OtherIndexTypes... dynvals) noexcept
-      : m_vals(static_cast<index_type>(dynvals)...) {}
+      : m_vals(static_cast<index_type>(dynvals)...) {
+#if MDSPAN_HAS_CXX_17
+    MDSPAN_IMPL_PRECONDITION(
+        detail::all_values_are_nonnegative_and_representable<index_type>(dynvals...));
+#endif
```

Comment:
Just noting that the precondition check will be performed *after* the data member is initialized.

Reply:
Yeah, I think it's too complex to check this before initialization -- this is also the approach libc++ takes so I think it's okay